### ISSUE TITLE
The next steps for the ModelWrapper

### ISSUE THREAD

**Issue by DomInvivo** [datamol-io/graphium#25]
In this thread, I want to discuss what are the next steps and the known problems for the `ModelWrapper` class, to make sure that the objectives are aligned and that we answer most of the problems.

**First step**

- [x] Clean-up molecular stuff using datamol
- [x] Change the name of the class to `PredictorModelWrapper` to better highlight that it is used for predictive models
- [x] Split the data loading from the model wrapper using `LightningDataModule`
- [ ] Make sure the data is transferred to the right device after the loading when the data is loaded from the CPU but the model runs on the GPU
- [x] Fix the metrics

**Second step**

- [x] Improve how the collate is done. Right now, a function `custom_collate` is used, but I think we can define a `collate_fn` in the data loader?
- [x] Implement `test_step` and `test_dataloader` in the `LightningDataModule`
- [ ] Make sure that the model wrapper allows to easily save and reload checkpoints

**Third step**

- [x] Check whether the class `BestEpochFromSummary` is still needed, or something similar is now in pytorch lightning
- [x]  Check whether the class `TrainingProgressFromSummary` is still needed, or something similar is now in pytorch lightning
- [x] Check whether the class `HyperparamsMetricsTensorBoardLogger` is still needed, or something similar is now in pytorch lightning
- [x] Right now, the number of workers for data loaders is fixed to 0 when using the GPU due to some bugs. Fix it to allow parallel data loading.
- [ ] Implement a better way to ignore nans from the target when computing the metrics and the loss, especially in multitask settings. 
- [x] Save final train/val/test performance and hparams in a YAML file, to avoid needing to open tensorboard to see them
